spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,654 @@
+    )
+    mandatory.add_argument(
+        '-metric',
+        required=False,
+        help='Metric to normalize. ',
+        default='diameter_AP',
+        choices=['diameter_AP', 'area', 'diameter_RL', 'eccentricity', 'solidity'],
+    )
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        '-normalize',
+        metavar=Metavar.int,
+        type=int,
+        choices=[0, 1],
+        default=1,  # Do we want 1 as default?
+        help='Set to 1 to normalize the metrics using a database of healthy controls. '
+    )
+    optional.add_argument(
+        '-extent',
+        type=float,
+        metavar=Metavar.float,
+        default=20.0,
+        help='Extent (in mm) to average metrics of healthy levels in superior-inferior direction.'
+    )
+    optional.add_argument(
+        '-distance',
+        type=float,
+        metavar=Metavar.float,
+        default=10.0,
+        help='Distance (in mm) in the superior-inferior direction from the compression to average healthy slices.'
+    )
+    optional.add_argument(
+        '-file-participants',
+        metavar=Metavar.file,
+        default='participants.tsv',
+        help='participants.tsv file of healthy controls. Use this flag in combination with the "-sex" and "-age" flags.'
+    )
+    optional.add_argument(
+        '-sex',
+        type=str,
+        choices=['F', 'M'],
+        help='Sex of healthy subject to use for the normalization. Requires the flag "-file-participants".'
+        ' By default, both sexes are used.'
+    )
+    optional.add_argument(
+        '-age',
+        type=int,
+        nargs=2,
+        metavar="[0 100]",
+        help='Age range of healthy subjects to use for the normalization. Requires the flag "-file-participants". '
+        'Example: "-age 60 80". By default, all ages are considered.'
+    )
+    optional.add_argument(
+        '-o',
+        metavar=Metavar.file,
+        help='Output CSV file name. If not provided, the suffix "_compression_metrics" is added to the file name provided by the flag "-i"'
+    )
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit")
+    optional.add_argument(
+        '-v',
+        metavar=Metavar.int,
+        type=int,
+        choices=[0, 1, 2],
+        default=1,
+        # Values [0, 1, 2] map to logging levels [WARNING, INFO, DEBUG], but are also used as "if verbose == #" in API
+        help="Verbosity. 0: Display only errors/warnings, 1: Errors/warnings + info messages, 2: Debug mode")
+
+    return parser
+
+
+# Functions for Step 1 (Load subject input files and get csv files)
+# ==========================================================================================
+def get_slice_thickness(img):
+    """
+    Get slice thickness from the input image.
+    :param img: Image: source image
+    :return float: slice thickness in mm.
+    """
+    return img.dim[5]
```

Comment:
Should this be [6] instead [5]?
```suggestion
    return img.dim[6]
```

Because, for instance, for axial image with slice thickness 3.6, `img.dim` is `(320, 320, 24, 1, 0.5, 0.5, 3.6000001, 1)`

Reply:
you are absolutly right! Thanks!